lamagraph-core/src/Core/Reducer.hs

```diff
@@ -0,0 +1,199 @@
+{-# LANGUAGE PartialTypeSignatures #-}
+{-# OPTIONS_GHC -Wno-partial-type-signatures #-}
```

Comment:
Why are these warnings disabled here?

Reply:
Because I want use partial type signature in [here](https://github.com/Lamagraph/interaction-nets-in-fpga/pull/21/files#diff-bf2ba3eb3bd77ba54822f5c1568d519b7400cb02b062deb1f9589d59fb87479cR90-R91).

Reply:
I'm not a fan of discarded signatures: you should use it iff you are prototyping without disabling warning. 
That's my vision, if you do not align with me, just resolve this comment.

Reply:
It is make your code more complex. I want use it just for telling some expression that length of `Vec` is `someNum`, I want to write `Vec someNum _` instead of `Vec someNum (Signal dom (Maybe (Node anotherNum)))`. I saw something like that [here](https://github.com/gergoerdi/retroclash-book-code/blob/master/src/led/blink/src/BlinkExplicitIndex.hs).

Simpler, I want to pass some concrete info into type signature and writing whole info may be confusing and nasty